R/EventStudy.R

```diff
@@ -170,56 +174,90 @@ EventStudy <- function(estimator, data, outcomevar, policyvar, idvar, timevar, c
     n_unique_rows <- nrow(data[!base::duplicated(data_ids),])
     if (n_unique_rows != n_units*n_periods) {
         warning("Dataset is unbalanced.")
+        unbalanced <- TRUE
+    } else {
+        unbalanced <- FALSE
+    }
+
+    data.table::setorderv(data, c(idvar, timevar))
+
+    detect_holes <- function(dt, idvar, timevar) {
+        dt <- data.table::as.data.table(dt)
+        holes_per_id <- dt[, .SD[!is.na(base::get(timevar))], by = c(idvar)
```

Comment:
Sure @nateschor!

```R
data.table::setorderv(data, c(idvar, timevar))  # Set timevar to be increasing within each id

detect_holes <- function(dt, idvar, timevar) {
  dt <- data.table::as.data.table(dt)
  holes_per_id <- dt[, .SD[!is.na(base::get(timevar))], by = c(idvar)         # Keep rows with nonmissing values in timevar
                 ][, list(holes = any(base::diff(base::get(timevar)) != 1)),  # Collapse by setting T if any difference in time-variable is not 1, and F otherwise
                      by = c(idvar)]                                          #    at the idvar level, so we have an id level dataset with TRUEs and FALSEs

return(any(holes_per_id$holes))   # Return T if any id has a hole
```

Does that help? I tried to use base functions to avoid calling new functions from packages beyond what we already use.

Reply:
The comments are really helpful, thanks @santiagohermo! Thanks for helping a `tidyverse` person learn some `data.table` :)